demisto_sdk/commands/common/clients/xsoar/xsoar_api_client.py

```diff
@@ -1343,3 +1421,134 @@ def update_playbook_input(self, playbook_id: str, new_inputs: dict):
             saving_inputs_path, method="POST", body={"inputs": new_inputs}
         )
         return self._process_response(response, status_code, 200)
+
+    def get_playbook_task_in_investigation(self, task_name, investigation_id):
+        tasks, status_code, _ = self._xsoar_client.generic_request(
+            f"/investigation/{investigation_id}/workplan/tasks", method="POST",
+            body={"states": self.PLAYBOOK_TASKS_STATES,
+                  "types": self.PLAYBOOK_TASKS_TYPES}, response_type="object"
+        )
+
+        for task in tasks:
+            if task_name == task.get('task').get('name'):
+                return task
+        raise ValueError(f'{task_name} task was not found in {investigation_id} investigation.')
+
+    def pull_playbook_tasks_by_state(self, incident_id: str,
+                                     task_input: str = None,
+                                     task_states: list = None,
+                                     task_name: str = None,
+                                     max_timeout: int = 60,
+                                     interval_between_tries: str = '3',
+                                     complete_task: bool = False,
+                                     ):
+        if not all(state in self.PLAYBOOK_TASKS_STATES for state in task_states):
+            raise ValueError(f'task_states are bad. Possible values: {self.PLAYBOOK_TASKS_STATES}')
+        if not task_states:
+            task_states = self.PLAYBOOK_TASKS_STATES
+        if complete_task and not task_input:
+            raise RuntimeError("Task input argument is missing to complete tasks.")
+        completed_tasks = []
+        found_tasks = []
+        start_time = time.time()
+
+        while True:
+            # Get all tasks with one state of the states in task_states list
+            tasks_by_states, status_code, _ = self._xsoar_client.generic_request(
```

Comment:
max_time is 60. is that seconds?
the loop will create multiple calls to this API 
I think we need to think about different solutions